src/mesh/boundary_info.C

```diff
@@ -212,6 +212,11 @@ void BoundaryInfo::regenerate_id_sets()
       _boundary_ids.insert(id);
       _shellface_boundary_ids.insert(id);
     }
+
+  // Handle global data
+  _communicator.set_union(_ss_id_to_name);
+  _communicator.set_union(_ns_id_to_name);
+  _communicator.set_union(_es_id_to_name);
```

Comment:
Could we make this a fixup on the previous commit?  The state in between is broken in a way that worries me, letting those maps get out of sync.

Reply:
done